pkg/porter/build.go

```diff
@@ -39,12 +39,38 @@ func (o *BuildOptions) Validate(cxt *context.Context) error {
 		o.Version = v.String()
 	}
 
-	return o.bundleFileOptions.Validate(cxt)
+	if o.Driver == "" {
+		o.Driver = p.Data.BuildDriver
+	} else {
+		if !stringSliceContains(BuildDriverAllowedValues, o.Driver) {
+			return errors.Errorf("invalid --driver value %s", o.Driver)
+		}
+
+		// Syncing value back to the config and we will always use the config
+		// to determine the driver
+		// This would be less awkward if we didn't do an automatic build during publish
+		p.Data.BuildDriver = o.Driver
+	}
+
+	return o.bundleFileOptions.Validate(p.Context)
+}
+
+func stringSliceContains(allowedValues []string, value string) bool {
```

Comment:
Should we move this to a more generic helpers file/location, or nah?

Reply:
I think we could use this in a couple places. I just was dithering about naming a package `util`, and imagined what people would think. 😂 

<img width="582" alt="Screen Shot 2021-05-14 at 8 05 57 AM" src="https://user-images.githubusercontent.com/1368985/118274819-3bb36b00-b48b-11eb-97e2-fd9079fe59c3.png">